ros2pkg/ros2pkg/resource/ament_cmake/CMakeLists.txt.em

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.5)
+project(@(project_name))
+
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+@[if cpp_library_name]@
+find_package(ament_cmake_ros REQUIRED)
+@[end if]@
+@[if dependencies]@
+@[  for deb in dependencies]@
+find_package(@deb REQUIRED)
+@[  end for]@
+
+@[  if cpp_library_name]@
+include_directories(
+  include
+)
+@[  end if]
+@[else]@
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> [REQUIRED] [QUIET])
+
+@[  if cpp_library_name]@
+include_directories(
+  include
+# include_directories($<dependency>_INCLUDE_DIRS})
+)
+@[  end if]@
+@[end if]@
+@[if cpp_library_name]@
+add_library(${PROJECT_NAME} src/@(cpp_library_name))
+
+@[  if dependencies]@
+ament_target_dependencies(
+  ${PROJECT_NAME}
+@[    for deb in dependencies]@
+  ${@(deb)_LIBRARIES}
+@[    end for]@
+)
+
+# Causes the visibility macros to use dllexport rather than dllimport,
+# which is appropriate when building the dll but not consuming it.
+target_compile_definitions(${PROJECT_NAME} PRIVATE "@(project_name.upper())_BUILDING_LIBRARY")
+@[  end if]@
+
+install(
+  DIRECTORY include/
+  DESTINATION include
+)
+install(
+  TARGETS ${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+)
+
+@[end if]@
+@[if cpp_node_name]@
+add_executable(${PROJECT_NAME}_node src/@(cpp_node_name))
+
+@[  if dependencies]@
+ament_target_dependencies(
+  ${PROJECT_NAME}_node
+@[    for deb in dependencies]@
+  ${@(deb)_LIBRARIES}
+@[    end for]@
+)
+
+@[  end if]@
+install(TARGETS ${PROJECT_NAME}_node
+  DESTINATION lib/${PROJECT_NAME})
+
+@[end if]@
+@[if cpp_library_name]@
+ament_export_include_directories(
+  include
+)
+ament_export_libraries(
+  ${PROJECT_NAME}
+)
+
+@[end if]@
+if(BUILD_TESTING)
+  find_package(ament_cmake_uncrustify REQUIRED)
+  ament_uncrustify()
+  # the lines above can be replaced with the lines below
+  # but it requires this repo to be a git repo in order
+  # to run all tests correctly
```

Comment:
when not a git repo, cpplint complains about the header guards not being format conform. It somehow interprets the full path of the file. When calling `git init` inside the package. THe headers are correct.
```
/Users/karsten/workspace/osrf/ros2_full/src/my_package/include/my_package/my_library.hpp:1:  #ifndef header guard has wrong style, please use: __USERS__KARSTEN__WORKSPACE__OSRF__ROS2_FULL__SRC__MY_PACKAGE__INCLUDE__MY_PACKAGE__MY_LIBRARY_HPP_  [build/header_guard] [5]
```

furthermore, copyright checks would also fail.

Reply:
I am not sure what it would take to makes these linters work "outside" of git. But I would suggest to still use ament auto and only disable the linters which have that problem with a comment mentioning why (see https://github.com/ros2/kdl_parser/blob/4151131ddd3339ab93bbf40bbe9cbc4fe1c6bc41/kdl_parser/CMakeLists.txt#L56 for an example). That way all other linters are still be used - including the ones potentially being added in the future.